### PR TITLE
Parse html indicators

### PR DESCRIPTION
**Pull request by melamedbn**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/38428

## Description
The ParseHTMLIndicators script will give customers the ability to parse given blogs/advisories with minimal false positives.
The main feature is the ability to parse domain indicators as is thanks to tld validation and exclusions.

## Minimum version of Cortex XSOAR
- [x] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

**Comment by shannon-holland** [comment]
@yaakovi Doc review completed.

File: Packs/CommonScripts/Scripts/ParseHTMLIndicators/ParseHTMLIndicators.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+script: ''
+type: python
+tags: []
+comment: This script extracts indicators from HTML and handles bad top-level domains to avoid false positives caused by file extensions.
+enabled: true
+args:
+- name: url
+  required: true
+  description: The full URL of the blog.
+- name: excludeIndicators
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/ParseHTMLIndicators/ParseHTMLIndicators.yml:14]: arguments should follow the snake_case convention

File: Packs/CommonScripts/Scripts/ParseHTMLIndicators/ParseHTMLIndicators.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+# Remove bad formatted indicators
+for fp_indicator in badDomainTLD:
+    blogIndicators[:] = (value for value in blogIndicators if value != fp_indicator)
+
+# Keep unique values
+blogIndicators = list(dict.fromkeys(blogIndicators))
+blogIndicators = str(blogIndicators)
+
+return_results(CommandResults(readable_output=blogIndicators, outputs={"http.parsedBlog.indicators": blogIndicators}))
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/ParseHTMLIndicators/ParseHTMLIndicators.py:74]: update the yml with the script outputs.

File: Packs/CommonScripts/Scripts/ParseHTMLIndicators/ParseHTMLIndicators.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
+import re
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/ParseHTMLIndicators/ParseHTMLIndicators.py:1]: please use functions, it's ready hard to follow the code this way.

**Review by yaakovi — COMMENTED** [review]
why is the new script in the `CommonScripts` pack?

@michalgold as discussed before, we should start thinking about limiting the use of the common packs as much as possible (large packs go against the marketplace concept).

**Comment by michalgold** [comment]
> why is the new script in the `CommonScripts` pack?
> 
> @michalgold as discussed before, we should start thinking about limiting the use of the common packs as much as possible (large packs go against the marketplace concept).

@yaakovi in general you are right, it is indeed the approach but this should be in common as it's going to be broadly used.

File: Packs/CommonScripts/Scripts/ParseHTMLIndicators/ParseHTMLIndicators.py (hunk 3 above)
Comment (melamedbn) [reply Packs/CommonScripts/Scripts/ParseHTMLIndicators/ParseHTMLIndicators.py:1]: I didn't use functions due to the type of the script and flow.
Will do an enhancement to the script with functions.

File: Packs/CommonScripts/Scripts/ParseHTMLIndicators/ParseHTMLIndicators.yml (hunk 1 above)
Comment (melamedbn) [reply Packs/CommonScripts/Scripts/ParseHTMLIndicators/ParseHTMLIndicators.yml]: Done.

File: Packs/CommonScripts/Scripts/ParseHTMLIndicators/ParseHTMLIndicators.py (hunk 2 above)
Comment (melamedbn) [reply Packs/CommonScripts/Scripts/ParseHTMLIndicators/ParseHTMLIndicators.py]: No outputs needed for this script.

Comment (yaakovi) [reply Packs/CommonScripts/Scripts/ParseHTMLIndicators/ParseHTMLIndicators.py]: you do have outputs:
`outputs={"http.parsedBlog.indicators": blogIndicators}`

therefore, `http.parsedBlog.indicators` should be declared as an output.

**Review by yaakovi — APPROVED** [review]
approving for now.
make sure to complete all previous comments

File: Packs/CommonScripts/Scripts/ParseHTMLIndicators/ParseHTMLIndicators.py (hunk 2 above)
Comment (melamedbn) [reply Packs/CommonScripts/Scripts/ParseHTMLIndicators/ParseHTMLIndicators.py]: Will fix that asap.

**Comment by melamedbn** [comment]
Regarding the ReDDoS issues detected:
Script uses common server regex variables.
